src/components/Trade/TradeInput.tsx

```diff
@@ -151,10 +155,15 @@ export const TradeInput = () => {
 
   const handleInputChange = useCallback(
     async (action: TradeAmountInputField, amount: string) => {
-      setValue('amount', amount)
-      setValue('action', action)
-      // If we've overridden the input we are no longer in sendMax mode
-      setValue('isSendMax', false)
+      swapperDispatch({
+        type: SwapperActionType.SET_VALUES,
+        payload: {
+          action,
+          // If we've overridden the input we are no longer in sendMax mode
```

Comment:
Not introduced by this PR but looks dangerous to me, shouldn't that be handled at reducer-level?